src/components/custom/_signup-affiliate-details.js

```diff
@@ -0,0 +1,398 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { Formik, Field } from 'formik'
+import { Flex } from '../containers'
+import {
+    getSignupAffiliateValue,
+    resetSignupAffiliateDetails,
+} from '../../pages/signup-affiliates-details/common/_utility'
+import { DropdownSearch } from '../elements'
+import { useResidenceList } from '../hooks/use-residence-list'
+import { useAffiliateData } from '../hooks/use-affiliate-data'
+import CurrencySelect from '../form/currency-select'
+import AgreementLabel from './_agreement-label'
+import { Input, Button } from 'components/form'
+import { Header, LinkText } from 'components/elements'
+import { localize, LocalizedLink } from 'components/localization'
+import device from 'themes/device.js'
+
+const StyledContentFlex = styled(Flex)`
+    width: 486px;
+    height: auto;
+    background-color: var(--color-white);
+    border-radius: 0.6rem;
+    box-shadow: 0 1.6rem 2rem 0 rgba(0, 0, 0, 0.1);
+    z-index: 1;
+    @media ${device.mobileL} {
+        width: 100%;
+        min-width: 360px;
+        padding: 40px 16px;
+    }
+`
+const Line = styled.div`
+    width: 126.5px;
+    height: 1px;
+    background-color: var(--color-grey-8);
+    @media ${device.mobileL} {
+        width: 97.5px;
+    }
+`
+const InputGroup = styled.div`
+    width: 100%;
+    position: relative;
+    margin: 40px 0 4px;
+    @media ${device.mobileL} {
+        margin: 32px 0 16px;
+    }
+`
+const SignupButton = styled(Button)`
+    width: 81px;
+    font-size: 14px;
```

Comment:
Button width should be hard coded, it should be dynamically calculated based on paddings. Font size defaults to 14px already.

Reply:
Removed hard coded values.